misc/python/materialize/checks/range.py

```diff
@@ -0,0 +1,110 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+from textwrap import dedent
+from typing import List
+
+from materialize.checks.actions import Testdrive
+from materialize.checks.checks import Check
+
+
+class Range(Check):
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view1;
+            > DROP MATERIALIZED VIEW IF EXISTS int_range_view2;
+            > DROP TABLE IF EXISTS int_range_table;
+            > CREATE TABLE int_range_table (f1 INT, f2 INT8RANGE);
+            > INSERT INTO int_range_table VALUES (1, '[2,16]'::INT8RANGE);
+            > INSERT INTO int_range_table VALUES (2, NULL);
+            """
+            )
+        )
+
+    def manipulate(self) -> List[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE MATERIALIZED VIEW int_range_view1 AS SELECT f1, f2 FROM int_range_table;
+                > INSERT INTO int_range_table VALUES (3, '[2,6)'::INT8RANGE);
+                > UPDATE int_range_table SET f2 = '[2,2)'::INT8RANGE WHERE f1 = 3;
```

Comment:
Yes, the `UPDATE` here is not needed because an UPDATE statement, unlike say a `CREATE TABLE` or `CREATE MATERIALIZED VIEW` statement, does not cause anything to be stored in the database catalog , and this test is about forcing stuff to be stored and then possibly upgraded and then read back.